clients/java/zpe/src/main/java/com/yahoo/athenz/zpe/AuthZpeClient.java

```diff
@@ -386,6 +386,24 @@ public static AccessCheckStatus allowAccess(String roleToken, String angResource
             if (rToken.validate(getZtsPublicKey(rToken.getKeyId()), allowedOffset, false, null) == false) {
                 LOG.error("allowAccess: Authorization denied. Authentication of token failed for token="
                         + rToken.getSignedToken());
+
+                // check the token expiration
+                long now    = System.currentTimeMillis() / 1000;
+                long expiry = rToken.getExpiryTime();
+                if (expiry != 0 && expiry < now) {
+                    String signedToken = rToken.getSignedToken();
+                    LOG.error("allowAccess: Authorization denied. Token expired. now=" +
+                            now + " expiry=" + expiry + " token=" + signedToken);
+                    try {
```

Comment:
This cache remove code is not necessary here since we know that the token is not in the cache. We only get within this if block if the entry is not in the cache.

Reply:
When we think about calling the method more than once, if the rToken in tokenCache expires, I believe we should remove the token from the cache since it's no longer valid.
Not only that, I thought this was necessary to keep the flow consistent since we also have it here:
https://github.com/yahoo/athenz/blob/4ddb8076490381099ea70468daed775215186857/clients/java/zpe/src/main/java/com/yahoo/athenz/zpe/AuthZpeClient.java#L441

Reply:
If the token is in the cache, then that code is not executed. If you look at the function it has:

// look in the cache
RoleToken rToken = null;

// if not in the cache 

if (roleToken == null) {
   // then we parse and validate
   // new code is within this block
}

the new code is inside that if block where we already determined that the token is not in the cache. So there is no point of calling to remove it from the cache.